internal/config/config.go

```diff
@@ -20,7 +20,8 @@ type Config struct {
 	NodeName       string        `mapstructure:"nodeName"`
 	HTTPServerPort string        `mapstructure:"httpServerPort"`
 	Interval       time.Duration `mapstructure:"interval"`
-	Timeout        time.Duration `mapstructure:"timeout"`
+	Timeout        time.Duration `mapstructure:"timeout"` // TODO: Unimplemented/unused
```

Comment:
would make sense to link it to the context of pester? 

If so in the config I would add a section like:
```
sink:
  HTTPServerPort:
  MaxRetries:
  Timeout:
```

Reply:
This is coupled to the issue Roberto pointed out. Maybe get rid of `maxRetries` and simply put a variable for the Context timeout directly and retry until we hit the context timeout (backoff time + connection time)

Do I reuse `timeout` for this? [In the comments in the chart](https://github.com/newrelic/helm-charts/blob/master/charts/newrelic-infrastructure-v3/values.yaml#L41-L42) this was meant to be the timeout to connect to the metrics we are scrapping.

I have no problem reusing `timeout` it but I have not enough context of this integration to see if there are any case where we would need two different timeouts.

Reply:
we should check in the old code what timeout was doing 😕

Reply:
Before it was honored. I'll create an issue to track that we lost that functionality

I do not like that much Timeout there, I would prefer a Timeout per component (ETCD, Kublect, KSM) and one for the Sink. 

Just one Timeout in the Top level is not that meaningful

Reply:
Good point.

What about this?
```yaml
sink:
  http:
    port:
    maxRetries:
    timeout:
```

There is no sight of using other sink than HTTP but this allows us to develop other sinks in the future (at config level).

Reply:
I like it 🆙

Reply:
Last commit added `sink` option tree.